### PR TITLE
add mailer for user

### PR DESCRIPTION
**Pull request by kuranovandry**

### CODE REVIEW

File: app/controllers/home_controller.rb [hunk 1]
```diff
@@ -5,4 +5,11 @@ class HomeController < ApplicationController
   def index
     @users = User.all
   end
+
+  def welcome
+    user = current_user
+    HomeMailer.welcome(user).deliver
+    redirect_to "/home"
```
Comment (maksymst) [inline app/controllers/home_controller.rb]: always use rails url helper

File: app/controllers/home_controller.rb [hunk 2]
```diff
@@ -5,4 +5,11 @@ class HomeController < ApplicationController
   def index
     @users = User.all
   end
+
+  def welcome
+    user = current_user
```
Comment (maksymst) [inline app/controllers/home_controller.rb]: redundant

File: app/views/home_mailer/welcome.html.haml [hunk 3]
```diff
@@ -0,0 +1 @@
+gfdgsdfgsdfgsdfg sdfg sdfgsd gsdfg sdf
```
Comment (maksymst) [inline app/views/home_mailer/welcome.html.haml]: use `Lorem ipsum...` for such cases

File: config/routes.rb [hunk 4]
```diff
@@ -2,6 +2,11 @@
   devise_for :admin_users, ActiveAdmin::Devise.config
   ActiveAdmin.routes(self)
   devise_for :users, :controllers => { :registrations => "registrations" }
-  resources :users
+  resources :users 
+  resources :home do
+    member do  
+      get 'welcome' => 'home#welcome'
```
Comment (maksymst) [inline config/routes.rb]: `get :welcome`  is enough

File: app/assets/javascripts/users.coffee [hunk 5]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (maksymst) [inline app/assets/javascripts/users.coffee]: remove

File: app/assets/stylesheets/users.scss [hunk 6]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Users controller here.
```
Comment (maksymst) [inline app/assets/stylesheets/users.scss]: remove

File: app/controllers/users_controller.rb [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+  before_action :authenticate_user!
+
+  def index
+    @users = User.all
+  end
+
+  def welcome
+    user = User.find(params[:id])
+    UserMailer.welcome(user).deliver
+    redirect_to :root
```
Comment (maksymst) [inline app/controllers/users_controller.rb]: why is it redirected to root?

File: app/controllers/users_controller.rb [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+
+  before_action :authenticate_user!
+
+  def index
+    @users = User.all
+  end
+
+  def welcome
+    user = User.find(params[:id])
+    UserMailer.welcome(user).deliver
```
Comment (maksymst) [inline app/controllers/users_controller.rb]: why is this not delayed?

File: config/routes.rb [hunk 9]
```diff
@@ -1,7 +1,12 @@
...
-  devise_for :users, :controllers => { :registrations => "registrations" }
-  resources :users
-  root :to => "home#index"
+  devise_for :users, :controllers => { :registrations => 'registrations' }
+  resources :users do
+    member do  
+      get 'welcome'
+    end 
+  end
+  resources :home
```
Comment (maksymst) [inline config/routes.rb]: I thought you removed homes controller

File: app/controllers/users_controller.rb (hunk 7 above)
Comment (kuranovandry) [inline app/controllers/users_controller.rb]: because root path is users#index

File: spec/controllers/users_controller_spec.rb [hunk 10]
```diff
@@ -0,0 +1,44 @@
+require 'rails_helper'
+
+RSpec.describe UsersController, type: :controller do
+  describe 'anonymous user' do
+    before :each do
+      login_with nil
+    end
+
+    it 'should be redirected to sign_in' do
```
Comment (maksymst) [inline spec/controllers/users_controller_spec.rb]: do not use `should`. 
correct description is `redirects to sign in page`

File: spec/controllers/users_controller_spec.rb [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+      get :index
+      expect( response ).to redirect_to( new_user_session_path )
```
Comment (maksymst) [inline spec/controllers/users_controller_spec.rb]: why do you need spaces before and after brackets?

File: spec/controllers/users_controller_spec.rb [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+    end
+  end
+  describe 'GET #index' do
```
Comment (maksymst) [inline spec/controllers/users_controller_spec.rb:7]: test for checking assigned variables is missing

File: spec/controllers/users_controller_spec.rb [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+
+    it 'should render index' do
+      login_with create( :user )
+      get :index
+      expect( response ).to render_template( :index )
+    end
+  end
+
+  describe 'create welcome email' do
+    it 'should deliver welcome email for user' do
```
Comment (maksymst) [inline spec/controllers/users_controller_spec.rb]: this test does nothing useful

**Pull request merged**